lib/src/native/realm_core.dart

```diff
@@ -157,10 +160,31 @@ class _RealmCore {
         _realmLib.realm_config_set_disable_format_upgrade(configHandle._pointer, config.disableFormatUpgrade);
       }
 
+      if (config.initialDataCallback != null) {
+        _realmLib.realm_config_set_data_initialization_function(
+            configHandle._pointer, Pointer.fromFunction(initial_data_callback, FALSE), config.toGCHandle());
+      }
+
       return configHandle;
     });
   }
 
+  static int initial_data_callback(Pointer<Void> userdata, Pointer<shared_realm> realmHandle) {
+    try {
+      final config = _realmLib.gc_handle_deref(userdata);
+      if (config is Configuration) {
```

Comment:
Maybe will have to delete this if statement after applying the suggestion above.